### PR TITLE
Use bare objects for the hacker news resources

### PR DESCRIPTION
**Pull request by schrockn**
### Summary & Motivation

I'm protoyping a bunch of resource and config API changes on the project fully featured. While doing that I'm discovering some easy wins that we can do now

### How I Tested These Changes

BK

### CODE REVIEW

**Comment by schrockn** [comment]
Current dependencies on/for this PR:
* master
  * **PR #11249** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11249" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/11249?utm_source=stack-comment).

**Review by yuhan — APPROVED** [review]

**Comment by schrockn** [comment]
**Graphite Merge Job**

Current status: 🔄 Waiting for checks to pass

This pull request is currently waiting for CI and mergeability checks to pass before being merged.

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/11249?utm_source=stack-comment).


Job Reference: VM0OdJSEhK9A4EbqgLZW

**Pull request merged**
